clients/src/main/java/org/apache/kafka/common/metrics/JmxReporter.java

```diff
@@ -160,17 +184,31 @@ private void reregister(KafkaMbean mbean) {
         }
     }
 
-    private static class KafkaMbean implements DynamicMBean {
+    public int size() {
+        return this.mbeans.size();
+    }
+
+    protected static class KafkaMbean implements DynamicMBean {
+        private final String metricName;
```

Comment:
It's impossible, as same `KafkaMbean` refers to several different `MetricName` objects, so key is calculated with a separate method:
```java
    protected String getMBeanName(MetricName metricName) {
...
    }
```
So I'm able to get rid of that questionable extra variable in favor of extra call of `getMBeanName`.

In fact, I'd prefer to do next updates:
1. Update the map as described above.
2. Hide `getMBeanName` into `KafkaMBean`.
3. Construct `KafkaMBean` with `MetricName` instead of `String`.